artifactory/services/go/publishzipandmod.go

```diff
@@ -34,29 +37,54 @@ func (pwa *publishZipAndModApi) isCompatible(artifactoryVersion string) bool {
 	return version.AtLeast(propertiesApi)
 }
 
-func (pwa *publishZipAndModApi) PublishPackage(params GoParams, client *jfroghttpclient.JfrogHttpClient, ArtDetails auth.ServiceDetails) error {
+func (pwa *publishZipAndModApi) PublishPackage(params GoParams, client *jfroghttpclient.JfrogHttpClient, ArtDetails auth.ServiceDetails) (*utils.OperationSummary, error) {
 	url, err := utils.BuildArtifactoryUrl(ArtDetails.GetUrl(), "api/go/"+params.GetTargetRepo(), make(map[string]string))
 	if err != nil {
-		return err
+		return nil, err
 	}
 	pwa.clientDetails = ArtDetails.CreateHttpClientDetails()
 	pwa.client = client
 	moduleId := strings.Split(params.GetModuleId(), ":")
+	totalSucceed, totalFailed := 0, 0
+	var filesDetails []clientutils.FileTransferDetails
 	// Upload zip file
-	err = pwa.upload(params.GetZipPath(), moduleId[0], params.GetVersion(), params.GetProps(), ".zip", url)
+	success, failed, err := uploadArchive(params, moduleId[0], ".zip", url, &filesDetails, pwa)
 	if err != nil {
-		return err
+		return nil, err
 	}
+	totalSucceed, totalFailed = totalSucceed+success, totalFailed+failed
 	// Upload mod file
-	err = pwa.upload(params.GetModPath(), moduleId[0], params.GetVersion(), params.GetProps(), ".mod", url)
+	success, failed, err = uploadArchive(params, moduleId[0], ".mod", url, &filesDetails, pwa)
 	if err != nil {
-		return err
+		return nil, err
 	}
+	totalSucceed, totalFailed = totalSucceed+success, totalFailed+failed
 	if version.NewVersion(pwa.artifactoryVersion).AtLeast(ArtifactoryMinSupportedVersionForInfoFile) && params.GetInfoPath() != "" {
 		// Upload info file. This supported from Artifactory version 6.10.0 and above
-		return pwa.upload(params.GetInfoPath(), moduleId[0], params.GetVersion(), params.GetProps(), ".info", url)
+		success, failed, err = uploadArchive(params, moduleId[0], ".info", url, &filesDetails, pwa)
+		totalSucceed, totalFailed = totalSucceed+success, totalFailed+failed
+		if err != nil {
+			return nil, err
+		}
 	}
-	return nil
+	tempFile, err := clientutils.SaveFileTransferDetailsInTempFile(&filesDetails)
+	if err != nil {
+		return nil, err
+	}
+	return &utils.OperationSummary{TotalSucceeded: totalSucceed, TotalFailed: totalFailed, TransferDetailsReader: content.NewContentReader(tempFile, "files")}, nil
+}
+
+func uploadArchive(params GoParams, moduleId, ext, url string, filesDetails *[]clientutils.FileTransferDetails, pwa *publishZipAndModApi) (success, failed int, err error) {
+	success, failed = 0, 1
+	details, err := pwa.upload(params.GetModPath(), moduleId, params.GetVersion(), params.GetProps(), ext, url)
+	if err != nil {
+		return
+	}
+	if details != nil {
```

Comment:
Do we actually need this if statement?
Wouldn't ```details``` be always not nil if we got here?